sdk/textanalytics/azure-ai-textanalytics/tests/test_analyze_sentiment.py

```diff
@@ -573,3 +574,92 @@ def callback(pipeline_response, deserialized, _):
+    def test_aspect_based_sentiment_analysis(self, client):
+        documents = [
+            "It has a sleek premium aluminum design that makes it beautiful to look at."
+        ]
+
+        document = client.analyze_sentiment(documents=documents, show_aspects=True)[0]
+
+        for sentence in document.sentences:
+            for aspect in sentence.aspects:
+                self.assertEqual('design', aspect.text)
+                self.assertEqual('positive', aspect.sentiment)
+                self.assertEqual(1.0, aspect.confidence_scores.positive)
+                self.assertEqual(0.0, aspect.confidence_scores.neutral)
+                self.assertEqual(0.0, aspect.confidence_scores.negative)
+                self.assertEqual(32, aspect.offset)
+                self.assertEqual(6, aspect.length)
+
+                sleek_opinion = aspect.opinions[0]
+                self.assertEqual('sleek', sleek_opinion.text)
+                self.assertEqual('positive', sleek_opinion.sentiment)
+                self.assertEqual(1.0, sleek_opinion.confidence_scores.positive)
+                self.assertEqual(0.0, sleek_opinion.confidence_scores.neutral)
+                self.assertEqual(0.0, sleek_opinion.confidence_scores.negative)
+                self.assertEqual(9, sleek_opinion.offset)
+                self.assertEqual(5, sleek_opinion.length)
+                self.assertFalse(sleek_opinion.is_negated)
+
+                premium_opinion = aspect.opinions[1]
+                self.assertEqual('premium', premium_opinion.text)
+                self.assertEqual('positive', premium_opinion.sentiment)
+                self.assertEqual(1.0, premium_opinion.confidence_scores.positive)
+                self.assertEqual(0.0, premium_opinion.confidence_scores.neutral)
+                self.assertEqual(0.0, premium_opinion.confidence_scores.negative)
+                self.assertEqual(15, premium_opinion.offset)
+                self.assertEqual(7, premium_opinion.length)
+                self.assertFalse(premium_opinion.is_negated)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    @TextAnalyticsClientPreparer(client_kwargs={"api_version": ApiVersion.V3_1_PREVIEW_1})
+    def test_aspect_based_sentiment_analysis_negated_opinion(self, client):
+        documents = [
+            "The food and service is not good"
+        ]
+
+        document = client.analyze_sentiment(documents=documents, show_aspects=True)[0]
+
+        for sentence in document.sentences:
+            food_aspect = sentence.aspects[0]
+            service_aspect = sentence.aspects[1]
+
+            self.assertEqual('food', food_aspect.text)
+            self.assertEqual('negative', food_aspect.sentiment)
+            self.assertEqual(0.01, food_aspect.confidence_scores.positive)
+            self.assertEqual(0.0, food_aspect.confidence_scores.neutral)
+            self.assertEqual(0.99, food_aspect.confidence_scores.negative)
```

Comment:
we don't have a good history of asserting equal on stuff like confidence scores... as the service updates the models we'll see slightly different scores causing the tests to fail unexpectedly.

Reply:
Sounds good, I will just assert that the positive and negative scores exist and that neutral is 0.0. Thanks!